src/adapters/RoomsJSONAdapter.js

```diff
@@ -73,10 +74,34 @@ export default class RoomsJSONAdapter extends RoomsAdapter {
   getRoomActivities(ID) {
     return Observable.create((observer) => {
       const data = !this.datasource[`${ID}-activities`] ? [] : this.datasource[`${ID}-activities`];
+      const formattedData = RoomsJSONAdapter.formatActivitiesData(data);
 
-      observer.next(data);
+      observer.next(formattedData);
 
       observer.complete();
     });
   }
 }
+
+/**
+ * Activities data could be strings of activity IDs or
+ * objects with metadata (like date separators).
+ * Convert the data into proper javascript objects
+ *
+ * @param {Array} activitiesData json formatted activity data
+ * @returns {Array}
+ */
+RoomsJSONAdapter.formatActivitiesData = (activitiesData) =>
+  activitiesData.map((activity) => {
+    if (typeof activity === 'object') {
+      // Clone the object since we will be modifying it
+      const formattedActivity = Object.assign({}, activity);
```

Comment:
2 part comment.
1) Wouldn't it be easier to just return the date object instead of an object that contains a date object? That way we avoid creating a lot of new objects.

2) If cloning the object is necessary, can you update this to use the spread operator? That's how it's being done in other places when objects are being copied.
```suggestion
      const formattedActivity = {...activity};
```

Alternatively, I actually think we can avoid this method altogether if we just return `{date: "string date"}` and have the `WebexActivityStream` make the object conversion just-in-time, since by the time we have to call `TimeRuler`, we know it has to be a date.
This would make our code a little more efficient by not looping through the whole array every time there's a render and potentially creating 2 new objects for non-activities. See my comment on `WebexActivityStream`

Reply:
2 part answer: 😄 

1. I'm following the requirements in the JIRA (https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-92114) that has the data shape form the adapter being `{date: 09/12/2019}`. Please update the requirements if you'd like it that way.

2. @akoushke commented the same thing. If there are consistent ways we want to do things, we need to add them as eslint rules (https://github.com/eslint/eslint/pull/9955 for this case). We can't expect contributors to be familiar with the entire codebase and to know how we expect certain patterns.

Reply:
@adamweeks, you're right, the requirements state that the adapter should return a `Date` object. The idea being, having some way of identifying when something is not an activity.

I realized that having that requirements makes the code unnecessarily and artificially inefficient. I will update the requirement for the adapter to have an object with a value string that is compliant with the [date-time string format](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#Date_Time_String_Format).
E.g. something like `{date: "2019-12-03"}`

Alternatively, we could flatten the array and only have strings and just validate whenever a string is a valid date with [datefns's `isValid`](https://date-fns.org/v1.9.0/docs/isValid)
e.g. `["2019-09-12". "activity1", "activity2" ...]`

Open for feedback

Reply:
I'm good with the data being a date string format as your example shows with: `{date: "2019-12-03"}`.

I prefer the data being passed to the components have a well defined shape. I think performing the datefn's validation on every item in the array on every render would be too much processing.

I'll update the PR to push the string to date conversion later in the process.